sklearn/neural_network/multilayer_perceptron.py

```diff
@@ -611,11 +641,24 @@ def fit(self, X, y):
             The target values (class labels in classification, real numbers in
             regression).
 
+        sample_weight : array-like, shape (n_samples,), optional (default=None)
+            Weights applied to individual samples (1. for unweighted).
+
+        class_weight : dict or 'balanced', optional
```

Comment:
Parameters to fit should be sample-wise. class_weight should be an estimator `__init__` parameter

Reply:
I see, you are right, `np.average` also validates the sizes for us.